src/extension/sldService/src/test/java/org/geoserver/sldservice/rest/ClassifierTest.java

```diff
@@ -2128,4 +2130,195 @@ private void checkNotOverlappingRules(Rule first, Rule second) throws IOExceptio
         assertTrue(first.getFilter().evaluate(feature));
         assertFalse(second.getFilter().evaluate(feature));
     }
+
+    @Test
+    public void testPercentagesInRulesLabelsVectors() throws Exception {
```

Comment:
Please split the tests in one test method per check, and validate the computed percentage values too.

Reply:
Please add a test for user provided classes as well, both raster and vector cases.
Also, these are all "happy paths", can you consider tests where things go wrong (e.g., rules that need to be merged, think of more) and add those as well?

Reply:
fixed